controllers/apidefinition_controller.go

```diff
@@ -284,6 +247,105 @@ func (r *ApiDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{RequeueAfter: queueA}, err
 }
 
+func (r *ApiDefinitionReconciler) processClientCertificateReferences(
+	ctx context.Context,
+	env *environmet.Env,
+	log logr.Logger,
+	upstreamRequestStruct *tykv1alpha1.ApiDefinition,
+) {
+	if len(upstreamRequestStruct.Spec.ClientCertificateRefs) != 0 {
+		clientCerts := make([]string, 0)
+
+		for _, secretName := range upstreamRequestStruct.Spec.ClientCertificateRefs {
+			tykCertID, err := r.checkSecretAndUpload(ctx, secretName, upstreamRequestStruct.Namespace, log, env)
+			if err != nil {
+				// we should log the missing secret, but we should still create the API definition
+				log.Info("failed to upload client certificate", "error", err)
```

Comment:
Should it be an error log (`log.Error`) or info?

Reply:
I kept it as Info because we still create API Definition.
I can make it more descriptive. Saying creating API without uploading this certificate maybe.

Reply:
makes sense 👍 let's keep it as it is.

Reply:
I changed it a bit. See if it is better.

Reply:
looks better! thank you